declcache.go

```diff
@@ -224,9 +225,23 @@ func build_package(p *build.Package) error {
 		log.Printf("package object: %s", p.PkgObj)
 		log.Printf("package source dir: %s", p.Dir)
 		log.Printf("package source files: %v", p.GoFiles)
+		log.Printf("GOPATH: %v", g_daemon.context.GOPATH)
+		log.Printf("GOROOT: %v", g_daemon.context.GOROOT)
 	}
+	env := os.Environ()
+	for i, v := range env {
+		if strings.HasPrefix(v, "GOPATH=") {
+			env[i] = "GOPATH=" + g_daemon.context.GOPATH
+		} else if strings.HasPrefix(v, "GOROOT=") {
```

Comment:
@shurcooL - it's a fair point. We should ensure both `GOROOT` and `GOPATH` are set.

@nsf - :+1: if it's any comfort, the feature works well for me!